### PR TITLE
New package: Gauche-0.9.9

### PR DESCRIPTION
**Pull request by rc-05**
**Description from [upstream](https://practical-scheme.net/gauche/index.html):**

Gauche is an R7RS Scheme implementation developed to be a handy script interpreter, which allows programmers and system administrators to write small to large scripts for their daily chores. Quick startup, built-in system interface, native multilingual support are some of my goals.

### CODE REVIEW

**Comment by Chocimier** [comment]
Cross compilation passes with Gauche added to hostmakedepends conditionally.

Conflicts with `gosh` package that installs `/usr/bin/gosh` too.

**Comment by rc-05** [comment]
> Cross compilation passes with Gauche added to hostmakedepends conditionally.
> 
> Conflicts with `gosh` package that installs `/usr/bin/gosh` too.

Now the application will not conflict anymore with the gosh package.

**Comment by rc-05** [comment]
> > Cross compilation passes with Gauche added to hostmakedepends conditionally.
> > Conflicts with `gosh` package that installs `/usr/bin/gosh` too.
> 
> Now the application will not conflict anymore with the gosh package and Gauche was added to hostmakedepends as you suggested.

**Comment by rc-05** [comment]
Let's see if Travis CI can build at least for the archs with glibc runtime.

**Comment by Chocimier** [comment]
> Let's see if Travis CI can build at least for the archs with glibc runtime.

I encourage you to look at build logs on travis to see what real problem is.

For cross compilation, problem is that gosh is executed during build. When it is native built, a while-ago created binary can be used. When cross-built, it cannot, but one can install native gosh by adding to hostmakedepends. 

But if you simply add Gauche to hostmakedepends, there is need for  Gauche to build Gauche. Solution is to only depend on Gauche in case of cross build, take a look for template of php.

**Comment by rc-05** [comment]
> > Let's see if Travis CI can build at least for the archs with glibc runtime.
> 
> I encourage you to look at build logs on travis to see what real problem is.
> 
> For cross compilation, problem is that gosh is executed during build. When it is native built, a while-ago created binary can be used. When cross-built, it cannot, but one can install native gosh by adding to hostmakedepends.
> 
> But if you simply add Gauche to hostmakedepends, there is need for Gauche to build Gauche. Solution is to only depend on Gauche in case of cross build, take a look for template of php.

Ok thanks! I'm going to follow your suggestions

**Comment by rc-05** [comment]
Can #21258, #21183 and #20739 be reviewed and merged (if everything goes smoothly)? @Chocimier

File: srcpkgs/Gauche/template [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+maintainer="rc-05 <rc23@email.it>"
+license="BSD-3-Clause"
+homepage="https://practical-scheme.net/gauche/"
+distfiles="${SOURCEFORGE_SITE}/gauche/${pkgname}-${version}.tgz"
+checksum=4ca9325322a7efadb9680d156eb7b53521321c9ca4955c4cbe738bc2e1d7f7fb
+alternatives="
+ scheme:scheme:/usr/bin/gauche
+ scheme:scheme.1:/usr/share/man/man1/gauche.1
+"
+nocross="Can't bootstrap with cross-compilation:"
```
Comment (Chocimier) [inline srcpkgs/Gauche/template:19]: Nope, it can be build.

File: srcpkgs/Gauche/template [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Template file for 'Gauche'
+pkgname=Gauche
+version=0.9.9
+revision=1
+build_style=gnu-configure
+configure_args="--prefix=/usr"
```
Comment (Chocimier) [inline srcpkgs/Gauche/template:6]: This is part of build-style, please remove.

File: srcpkgs/Gauche/template [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+hostmakedepends="gzip texinfo"
```
Comment (Chocimier) [inline srcpkgs/Gauche/template:7]: gzip is part of base-chroot, please remove.

**Comment by rc-05** [comment]
Good news: it seems that it is now cross compiling without any problem...
I suspect that the cross compilation problems were generated by some explicit `hostmakedepends` and `configure_args` flags that are now automatically resolved when left implicit.

**Comment by rc-05** [comment]
Going to amend the commit as soon as it finished cross-compiling locally.

**Comment by rc-05** [comment]
I think that I've resolved the renaming issue... Compiling locally to see if the cross package is generated.

**Comment by rc-05** [comment]
I think that for some reason at install point the `gauche` interpreter is not found... Maybe the internal makefile contains an undocumented flag/variable or even worse it's hardcoded and a patch is needed..

I'm surprisingly having an hard time packaging Gauche: should I mark this pull request as WIP at this point? @Chocimier

**Comment by rc-05** [comment]
Marked as [WIP] as I am trying to figure out how to successfully resolve the binary conflict with Gosh and Cross Compiling

**Comment by Chocimier** [comment]
You can just set `conflicts` variable if installing Gauche aside of gosh is ok for you, with short comment explaining reason.

**Comment by Chocimier** [comment]
*forbidding installing

**Comment by rc-05** [comment]
I could have done that and call it a day but maybe there are people that use both Gosh and Gauche for some reason or another so there's that... And maybe I think that I've finally resolved the naming conflict after days and days...

**Comment by Chocimier** [comment]
Much better now.

Conditional post_install is, fortunately, overwritten by other one, manpage name should be same whether cross-compiled or not. Please remove.

Is libatomic_ops-devel needed, or not?

`--host` is already set bu build style.

Install messages are mentioned for more important things, renamed binary can be known from `xbps-query -f`.

**Comment by rc-05** [comment]
libatomic seems to cause problems for cross compilation (it is not mandatory, maybe it's best that I remove it)

**Comment by rc-05** [comment]
The libatomic library is not mandatory: for simplicity's sake I've removed it from the template.
Should instead be added as a build option?

**Comment by Chocimier** [comment]
As you wish. I think it's ready to merge.

**Comment by rc-05** [comment]
Hmmm then I'll add it as a build option and push the commit asap

**Comment by rc-05** [comment]
Build option added: I also think that this package can be finally merged into master repo of void-packages

**Comment by Chocimier** [comment]
I'm still unsure if libatomic should be option or enabled based on $XBPS_TARGET_NO_ATOMIC8, or both.

**Comment by rc-05** [comment]
I think that it would be better to leave it as a build option since it is not mandatory

**Comment by Chocimier** [comment]
Documentation says, that bundled libatomic is used, if needed and not found in system. Remove build option and add libatomic to makedepends unconditionally.

**Pull request merged**